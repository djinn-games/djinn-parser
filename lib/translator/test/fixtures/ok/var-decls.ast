{
  "type": "Program",
  "name": {
    "type": "Identifier",
    "name": "var_declarations",
    "line": 1
  },
  "body": {
    "type": "Body",
    "sentences": [
      {
        "type": "ExpressionSentence",
        "expression": {
          "type": "VarDeclaration",
          "dataType": "int",
          "id": {
            "type": "Identifier",
            "name": "x",
            "line": 3
          },
          "line": 3
        },
        "line": 3
      },
      {
        "type": "ExpressionSentence",
        "expression": {
          "type": "VarDeclaration",
          "dataType": "float",
          "id": {
            "type": "Identifier",
            "name": "y",
            "line": 4
          },
          "init": {
            "type": "VarDeclaration",
            "dataType": "float",
            "id": {
              "type": "Identifier",
              "name": "z",
              "line": 4
            },
            "init": {
              "type": "Literal",
              "line": 4,
              "dataType": "float",
              "value": 3.5
            },
            "line": 4
          },
          "line": 4
        },
        "line": 4
      },
      {
        "type": "ExpressionSentence",
        "expression": {
          "type": "VarDeclaration",
          "dataType": "bool",
          "id": {
            "type": "Identifier",
            "name": "flag",
            "line": 5
          },
          "init": {
            "type": "Literal",
            "line": 5,
            "dataType": "bool",
            "value": false
          },
          "line": 5
        },
        "line": 5
      },
      {
        "type": "ExpressionSentence",
        "expression": {
          "type": "VarDeclaration",
          "dataType": "str",
          "id": {
            "type": "Identifier",
            "name": "txt",
            "line": 6
          },
          "line": 6
        },
        "line": 6
      }
    ],
    "line": 2
  },
  "line": 1
}