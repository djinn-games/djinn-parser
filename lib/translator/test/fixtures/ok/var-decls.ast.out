{
  "type": "Program",
  "body": [
    {
      "type": "ExpressionStatement",
      "expression": {
        "type": "AssignmentExpression",
        "operator": "=",
        "left": {
          "type": "MemberExpression",
          "object": {
            "type": "MemberExpression",
            "object": {
              "type": "Identifier",
              "name": "DJINN"
            },
            "property": {
              "type": "Identifier",
              "name": "__scope"
            }
          },
          "property": {
            "type": "Literal",
            "value": "_x",
            "raw": "\"_x\""
          }
        },
        "right": {
          "type": "Literal",
          "value": 0,
          "raw": "0"
        }
      }
    },
    {
      "type": "ExpressionStatement",
      "expression": {
        "type": "AssignmentExpression",
        "operator": "=",
        "left": {
          "type": "MemberExpression",
          "object": {
            "type": "MemberExpression",
            "object": {
              "type": "Identifier",
              "name": "DJINN"
            },
            "property": {
              "type": "Identifier",
              "name": "__scope"
            }
          },
          "property": {
            "type": "Literal",
            "value": "_y",
            "raw": "\"_y\""
          }
        },
        "right": {
          "type": "AssignmentExpression",
          "operator": "=",
          "left": {
            "type": "MemberExpression",
            "object": {
              "type": "MemberExpression",
              "object": {
                "type": "Identifier",
                "name": "DJINN"
              },
              "property": {
                "type": "Identifier",
                "name": "__scope"
              }
            },
            "property": {
              "type": "Literal",
              "value": "_z",
              "raw": "\"_z\""
            }
          },
          "right": {
            "type": "Literal",
            "value": 3.5,
            "raw": "3.5"
          }
        }
      }
    },
    {
      "type": "ExpressionStatement",
      "expression": {
        "type": "AssignmentExpression",
        "operator": "=",
        "left": {
          "type": "MemberExpression",
          "object": {
            "type": "MemberExpression",
            "object": {
              "type": "Identifier",
              "name": "DJINN"
            },
            "property": {
              "type": "Identifier",
              "name": "__scope"
            }
          },
          "property": {
            "type": "Literal",
            "value": "_flag",
            "raw": "\"_flag\""
          }
        },
        "right": {
          "type": "Literal",
          "value": false,
          "raw": "false"
        }
      }
    }
  ]
}
